test/unit_tests/arkode/gtest/CMakeLists.txt

```diff
@@ -0,0 +1,20 @@
+# ---------------------------------------------------------------
+# SUNDIALS Copyright Start
+# Copyright (c) 2002-2023, Lawrence Livermore National Security
+# and Southern Methodist University.
+# All rights reserved.
+#
+# See the top-level LICENSE and NOTICE files for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+# SUNDIALS Copyright End
+# ---------------------------------------------------------------
+
+# include location of public and private header files
+
+add_executable(test_arkode_error_handling test_arkode_error_handling.cpp)
+target_include_directories(test_arkode_error_handling PRIVATE
+  $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/include> ${CMAKE_SOURCE_DIR}/include ${CMAKE_SOURCE_DIR}/src)
+target_link_libraries(test_arkode_error_handling
+    PRIVATE  sundials_arkode sundials_nvecserial sundials_core GTest::gtest_main GTest::gmock)
```

Comment:
These GTest targets require CMake 3.23

Reply:
So that seems to only be true if you use `find_package(GTest)`. Right now we try `find_package(GTest)` first, and if that fails then we follow the recommended way of including GoogleTest, which is to use the `FetchContent` approach. The `FetchContent` approach does result in the `GTest::gtest_main GTest::gmock` targets both being available. As such, I have removed our call to `find_package(GTest)` and we can use these targets as it becomes independent of the CMake version.